### PR TITLE
Mitigate to .Net 5.

### PR DESCRIPTION
**Pull request by mahoshojoHCG**
What have changed:
- .Net Framework to .Net 5
- Code shared project to new .Net Standard project
- All obsoleted APIs either removed, ported to the latest version or found replacement
- Code Analysis added

Act almost as the same as dev branch.

Known issues:
- Maximum window will pump a error window.
- Black startup background screen, theme not working on main window.
- More than 1,000 warnings due to Code Analysis

### CODE REVIEW

**Comment by mahoshojoHCG** [comment]
The ci needs update to accept .Net 5 test.

**Comment by mahoshojoHCG** [comment]
The latest commit contains GitHub Actions.

File: src/NsisoLauncher/Config/ConfigHandler.cs [hunk 1]
```diff
@@ -1,100 +1,100 @@
...
+using Newtonsoft.Json.Linq;
+using NsisoLauncher.Utils;
+using NsisoLauncherCore;
+using NsisoLauncherCore.Modules;
+using NsisoLauncherCore.Net;
 
 namespace NsisoLauncher.Config
 {
-    public class ConfigHandler
+    public class ConfigHandler : IDisposable
```
Comment (Nsiso) [inline src/NsisoLauncher/Config/ConfigHandler.cs:16]: This is sweet ;)

File: src/NsisoLauncherCore/Auth/IAuthenticator.cs [hunk 2]
```diff
@@ -26,11 +29,6 @@ public class AuthenticateResult
...
 
         public UserData UserData { get; set; }
-
-        public AuthenticateResult(AuthState state)
-        {
-            this.State = state;
-        }
     }
 
     public enum AuthState
```
Comment (Nsiso) [inline src/NsisoLauncherCore/Auth/IAuthenticator.cs:34]: According to the GitHub action's warning, you should remove the underscores from member name NsisoLauncherCore.Auth.AuthState.
I don't know why they suggest doing this, but I think it is better to fix it.

File: src/NsisoLauncherCore/Auth/YggdrasilAuthenticator.cs [hunk 3]
```diff
@@ -123,144 +88,53 @@ public class YggdrasilTokenAuthenticator : IAuthenticator
...
-            catch (Exception ex)
-            {
-                return new AuthenticateResult(AuthState.ERR_INSIDE)
-                {
-                    Error = new Net.MojangApi.Error() { ErrorMessage = ex.Message, Exception = ex },
-                    AccessToken = AccessToken = this.AccessToken,
-                    UserData = this.UserData,
-                    SelectedProfileUUID = this.SelectedProfileUUID
-                };
-            }
```
Comment (Nsiso) [inline src/NsisoLauncherCore/Auth/YggdrasilAuthenticator.cs:193]: This code is used to refresh the Access token, should not be deleted.

File: src/NsisoLauncherCore/Auth/YggdrasilAuthenticator.cs [hunk 4]
```diff
@@ -1,118 +1,83 @@
...
+using System.Net;
 using System.Threading.Tasks;
+using NsisoLauncherCore.Net.MojangApi;
+using NsisoLauncherCore.Net.MojangApi.Api;
+using NsisoLauncherCore.Net.MojangApi.Endpoints;
 using static NsisoLauncherCore.Net.MojangApi.Responses.AuthenticateResponse;
 
 namespace NsisoLauncherCore.Auth
 {
     public class YggdrasilAuthenticator : IAuthenticator
```
Comment (Nsiso) [inline src/NsisoLauncherCore/Auth/YggdrasilAuthenticator.cs:12]: Better to rewrite it.

File: src/NsisoLauncherCore/Util/AssetsReader.cs [hunk 5]
```diff
@@ -44,16 +44,18 @@ public JAssets GetAssets(Modules.Version version)
 
     public class JAssets
     {
-        [JsonProperty("objects")]
-        public Dictionary<string, JAssetsInfo> Objects { get; set; }
+        [JsonProperty("objects")] public Dictionary<string, JAssetsInfo> Objects { get; set; }
```
Comment (Nsiso) [inline src/NsisoLauncherCore/Util/AssetsReader.cs:47]: Github action's warning:
Change 'Objects' to be read-only by removing the property setter.

File: src/NsisoLauncherCore/Util/SystemTools.cs [hunk 6]
```diff
@@ -16,43 +16,60 @@ public enum ArchEnum
...
+                        return 4096;
+                    else
+                        return rm;
+
+                default:
+                    return rm;
             }
-            else
+
+            return 1024;
```
Comment (Nsiso) [inline src/NsisoLauncherCore/Util/SystemTools.cs:46]: Unreachable code detected.

File: src/NsisoLauncher/Views/Windows/NewDownloadTaskWindow.xaml.cs [hunk 7]
```diff
@@ -204,44 +161,16 @@ private async void DownloadForgeButton_Click(object sender, RoutedEventArgs e)
...
-            JWLiteloader liteloader = null;
-            if (liteloaderListDataGrid.SelectedItem != null)
-            {
-                liteloader = (JWLiteloader)liteloaderListDataGrid.SelectedItem;
-            }
-            else
-            {
-                await this.ShowMessageAsync("您未选择要安装的Liteloader", "您需要选择一个要安装Liteloader");
-                return;
-            }
```
[the remark this answers is not in the document]
Comment (mahoshojoHCG) [reply src/NsisoLauncher/Views/Windows/NewDownloadTaskWindow.xaml.cs:241]: Deleted code was unaccessable.

[the remark this answers is not in the document]
Comment (mahoshojoHCG) [reply src/NsisoLauncher/Views/Windows/NewDownloadTaskWindow.xaml.cs:241]: Deleted code was unaccusable.

File: src/NsisoLauncherCore/Auth/YggdrasilAuthenticator.cs (hunk 3 above)
Comment (mahoshojoHCG) [reply src/NsisoLauncherCore/Auth/YggdrasilAuthenticator.cs:193]: The code resolver auto merge some same state to avoid nesting.

**Pull request merged**